### ISSUE TITLE
Misleading manual_memcpy suggestion

### ISSUE THREAD

**Issue by rizakrko** [rust-lang-nursery/rust-clippy#3004]
Following code:
```rust
fn copy(arr: &mut [u8], from: usize, target: &[u8]) {
    for i in from..from + target.len() {
        arr[i] = target[i - from];
    }
}
```
triggers this warning:
```rust
warning: it looks like you're manually copying between slices
 --> src/main.rs:2:14
  |
2 |     for i in from..from + target.len() {
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing the loop by: `arr[from..from + target.len()].clone_from_slice(&target[(from - from)..(from + target.len() - from)])`
  |
  = note: #[warn(manual_memcpy)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#manual_memcpy
```
But this contains `(from - from)`, which by itself leads to next error:
```rust
error: equal expressions as operands to `-`
 --> src/main.rs:3:35
  |
3 |         .clone_from_slice(&target[(from - from)..(from + target.len() - from)])
  |                                   ^^^^^^^^^^^^^
  |
  = note: #[deny(eq_op)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#eq_op
```
In the case of copying the entire slice `clippy` may provide specialized  suggestion
```rust
warning: it looks like you're manually copying between slices
 --> src/main.rs:2:14
  |
2 |     for i in from..from + target.len() {
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing the loop by: `arr[from..from + target.len()].clone_from_slice(&target)`
  |
```

**Comment by mikerite** [comment]
Related to #2828
